src/AWS/Shared/Storage/DynamoDBStorage.cs

```diff
@@ -163,34 +165,143 @@ private async Task CreateTable(string tableName, List<KeySchemaElement> keys, Li
 
             try
             {
-                var response = await ddbClient.CreateTableAsync(request);
-                TableDescription description = null;
-                do
+                await ddbClient.CreateTableAsync(request);
+                TableDescription description = await TableWaitOnStatusAsync(tableName, TableStatus.CREATING, TableStatus.ACTIVE);
+                await TableUpdateTtlAsync(tableName, ttlAttributeName);
+            }
+            catch (Exception exc)
+            {
+                Logger.Error(ErrorCode.StorageProviderBase, $"Could not create table {tableName}", exc);
+                throw;
+            }
+        }
+
+        private async Task TableUpdateTtlAsync(string tableName, string ttlAttributeName)
+        {
+            if (!string.IsNullOrEmpty(ttlAttributeName))
+            {
+                return;
+            }
+
+            await ddbClient.UpdateTimeToLiveAsync(new UpdateTimeToLiveRequest
```

Comment:
Could check the existing ttl attribute on the table. Since it requires a request, hard to say if it's any better though.

Reply:
I think it's fine to keep as is.